### ISSUE TITLE
TrabalhoRobutler missing items

### ISSUE THREAD

**Issue by Bereke23** [miguelriemoliveira/psr_22-23#12]
![gazebo_screenshot](https://user-images.githubusercontent.com/101461378/212198500-dc90b564-e5a1-441b-91b0-1cfc047cf96b.png)
After installing the two aws repos in the catkin_ws, as instructed on part 12 ex8, we are still missing some items. There is a screenshot of our gazebo. Please, let us know how we can fix it.

**Comment by miguelriemoliveira** [comment]
Which items are missing?

**Comment by Tisila** [comment]
If we take a look at the original environment from the class repository below.
Some of the items that are missing:
- desk in the office
- coffee machine in the kitchen
- flower pot in the balcony
- vases throughout the house
- sink in the kitchen

![original_environment](https://user-images.githubusercontent.com/35962520/212205540-00fe2363-b9bd-40f6-8af3-726a479594e8.png)
)

**Comment by miguelriemoliveira** [comment]
Hum, did you compile your catkin workspace after downloading both aws repositories to your catkin_ws/src folder?

**Comment by hugohashi** [comment]
Yes, we did, but it didn't work.

**Comment by miguelriemoliveira** [comment]
Ok, my suggestion is to meet tomorrow night by zoom. At 21h is ok for you?

**Comment by Tisila** [comment]
Yes, we are available in that time window, Tuesday 17th at 21h00, to troubleshoot this issue.

**Comment by Tisila** [comment]
We are connected to the "PSR - Aula T" Zoom meeting

**Comment by miguelriemoliveira** [comment]
Hi ,

Please give me 5 mins to login...

**Comment by Tisila** [comment]
```
[Err] [InsertModelWidget.cc:403] Missing model.config for model "/home/tisila/Documents/PSR/catkin_ws/src/turtlebot3_simulations/turtlebot3_gazebo/models/turtlebot3_autorace"
[Err] [InsertModelWidget.cc:403] Missing model.config for model "/home/tisila/Documents/PSR/catkin_ws/src/turtlebot3_simulations/turtlebot3_gazebo/models/turtlebot3_autorace_2020"
[Err] [InsertModelWidget.cc:403] Missing model.config for model "/home/tisila/Documents/PSR/catkin_ws/src/turtlebot3_simulations/turtlebot3_gazebo/models/turtlebot3_common_meshes"

```

**Comment by Tisila** [comment]
```
[Wrn] [SystemPaths.cc:459] File or path does not exist [""] [model://Toilet/meshes/toilet.obj]
[Err] [Visual.cc:3030] No mesh specified
```

**Comment by Goncalo287** [comment]
We're having the same problem, did you find a fix for it?

**Comment by Tisila** [comment]
## What we tried

We have concluded that the problem comes from the fact that there are models missing from the directory paths defined in `$GAZEBO_MODEL_PATH` environment variable.

We added the following code block to our `~/.bashrc` to add known models directories to the environment variable.
```bash
export GAZEBO_MODEL_PATH="~/.gazebo/models:${GAZEBO_MODEL_PATH}"
export GAZEBO_MODEL_PATH="~/catkin_ws/src/psr_22-23/TrabalhoRobutler/psr_apartment_description/description/models:${GAZEBO_MODEL_PATH}"
```

We tried to find the missing models online, we even downloaded the models from [gazebo_models](https://github.com/osrf/gazebo_models) to the `~/.gazebo/models` but it didn't work... We noticed however that the missing models from Gazebo are not in our system.

If you want to debug your Gazebo further, add a variable `<arg name="verbose" value="true"/>` to enable verbose mode. This variable should be placed in your `gazebo.launch` file inside the following structure:
```xml
<launch>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find psr_apartment_description)/worlds/psr_apartment.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
    <arg name="verbose" value="true"/>
  </include>
</launch>
```
Set verbose to false once you're done since it's a lot of output.

## The solution

The solution that the teacher gave us was to edit the `gazebo.launch` to load an empty apartment with 
```xml
<arg name="world_name" value="$(find psr_apartment_description)/worlds/apartment_empty.world"/>
```
and then populate the house (just like in the Sims game) in Gazebo GUI with the items available in the `Insert` tab on the left side panel by clicking the item on the list and then clicking on the map to place it. After placing all the furniture, click `File > Save World As` and save it inside your ros-package. Finally, edit the gazebo.launch again to point the `worlds_name` variable to your newly created world.

**Comment by miguelriemoliveira** [comment]
Thanks @Tisila for the detailed explanation.
